gcs/tests/gcs-sample.yaml

```diff
@@ -0,0 +1,152 @@
+              workspace: credentials
+            params:
+            - name: bucketName
+              value: gs://tekton-test-bucket-$(uid)
+            - name: project
+              value: $(params.project-name)
+          - name: create-data
+            runAfter: [gcs-create-bucket]
+            taskSpec:
+              workspaces:
+              - name: data
+              steps:
+              - name: write-data
+                image: ubuntu
+                script: |
+                  #!/usr/bin/env bash
+                  set -xe
+                  mkdir -p $(workspaces.data.path)/$(uid)/cookies/
+                  echo "Test data $(uid)" > $(workspaces.data.path)/$(uid)/cookies/recipe
+            workspaces:
+            - name: data
+              workspace: data
+          - name: gcs-upload
+            taskRef:
+              name: gcs-upload
+            runAfter: [create-data]
+            workspaces:
+            - name: credentials
+              workspace: credentials
+            - name: source
+              workspace: data
+            params:
+            - name: path
+              value: $(uid)
+            - name: location
+              value: gs://tekton-test-bucket-$(uid)
+            - name: serviceAccountPath
+              value: service_account.json
+          - name: gcs-download
+            taskRef:
+              name: gcs-download
+            runAfter: [gcs-upload]
+            workspaces:
+            - name: credentials
+              workspace: credentials
+            - name: output
+              workspace: data
+            params:
+            - name: name
+              value: download-$(uid)
+            - name: location
+              value: gs://tekton-test-bucket-$(uid)
+            - name: typeDir
+              value: "true"
+            - name: serviceAccountPath
+              value: service_account.json
+          - name: verify-data
+            runAfter: [gcs-download]
+            workspaces:
+            - name: data
+              workspace: data
+            taskSpec:
+              workspaces:
+              - name: data
+              steps:
+              - image: ubuntu
+                script: |
+                  #!/usr/bin/env bash
+                  set -xe
+                  ls -R $(workspaces.data.path)
+                  cat $(workspaces.data.path)/download-$(uid)/cookies/recipe | grep "Test data $(uid)"
+          - name: delete-bucket
+            taskRef:
+              name: gcs-delete-bucket
+            runAfter: [gcs-download]
+            workspaces:
+            - name: credentials
+              workspace: credentials
+            params:
+            - name: bucketName
+              value: gs://tekton-test-bucket-$(uid)
+---
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: gcs-binding
+spec:
+  params:
+    - name: pvc-name
+      value: gcs-test-storage
+    - name: project-name
+      value: hebridean-sun
```

Comment:
oops.  obv this will need to not be hard coded in future...

Reply:
I think the next step here will be to setup the logic that creates the TriggerBindings and Eventlistener on command (and cleans them up).

@sbwsg if we want to actually merge the PR one option is to remove the trigger binding + event listener and merge the TriggerTemplate as is (kinda like what i did with the pull request Task)

Reply:
I've done as suggested and reduced this diff to just the TriggerTemplate.